includes/class-newspack-newsletters.php

```diff
@@ -281,11 +333,132 @@ public static function display_post_states( $post_states, $post ) {
+	 *
+	 * @param array $query The WP query object.
+	 */
+	public static function maybe_display_public_archive_posts( $query ) {
+		if ( is_admin() || ! $query->is_main_query() || ! is_post_type_archive( self::NEWSPACK_NEWSLETTERS_CPT ) ) {
+			return;
+		}
+
+		$meta_query = $query->get( 'meta_query' );
+
+		if ( empty( $meta_query ) || ! is_array( $meta_query ) ) {
+			$meta_query = [];
+		}
+
+		$meta_query[] = [
+			'key'          => 'is_public',
+			'value'        => '1',
+			'meta_compare' => '=',
+		];
+
+		$query->set( 'meta_query', $meta_query );
+	}
+
+	/**
+	 * Decide whether this newsletter should be publicly viewable as a page.
+	 * Triggers a 404 if the current page is a single Newsletter and not marked public.
+	 */
+	public static function maybe_display_public_post() {
+		if (
+			current_user_can( 'edit_others_posts' ) ||
+			! is_singular( self::NEWSPACK_NEWSLETTERS_CPT )
+		) {
+			return;
+		}
+
+		$is_public = get_post_meta( get_the_ID(), 'is_public', true );
+
+		// If not marked public, make it a 404 to non-logged-in users.
+		if ( empty( $is_public ) ) {
+			global $wp_query;
+
+			// Replace document title with 'Page not found'.
+			add_filter(
+				'wpseo_title',
+				function( $title ) {
+					return str_replace( get_the_title(), __( 'Page not found', 'newspack-newsletters' ), $title );
+				}
+			);
+
+			status_header( 404 );
+			nocache_headers();
+			include get_query_template( '404' );
+			die();
+		}
+	}
+
+	/**
+	 * Make "View" links say "Preview" if the newsletter is not marked as public.
+	 *
+	 * @param array $actions Array of action links to be shown in admin posts list.
+	 * @return array Filtered array of action links.
+	 */
+	public static function display_view_or_preview_link_in_admin( $actions ) {
+		if ( 'publish' !== get_post_status() || self::NEWSPACK_NEWSLETTERS_CPT !== get_post_type() ) {
+			return $actions;
+		}
+
+		$is_public = get_post_meta( get_the_ID(), 'is_public', true );
+
+		if ( empty( $is_public ) && isset( $actions['view'] ) ) {
+			$actions['view'] = '<a href="' . esc_url( get_the_permalink() ) . '" rel="bookmark" aria-label="View ' . esc_attr( get_the_title() ) . '">Preview</a>';
+		}
+
+		return $actions;
+	}
+
+	/**
+	 * Disable auto-saves after newsletter has been sent.
+	 */
+	public static function maybe_disable_autosave() {
+		if ( 'publish' !== get_post_status() || self::NEWSPACK_NEWSLETTERS_CPT !== get_post_type() ) {
+			return false;
+		}
+
+		wp_dequeue_script( 'autosave' );
+	}
+
+	/**
+	 * Disable Newspack Campaigns on Newsletter posts.
+	 *
+	 * @param array $disabled Disabled status to filter.
+	 * @return array|boolean Unfiltered disabled status, or true to disable.
+	 */
+	public static function disable_campaigns_for_newsletters( $disabled ) {
+		if ( self::NEWSPACK_NEWSLETTERS_CPT === get_post_type() ) {
+			return true;
+		}
+
+		return $disabled;
+	}
+
+	/**
+	 * Disable Jetpack Related Posts on Newsletter posts.
```

Comment:
Maybe I'm missing something here, but why would recirculation here be unwanted? Or is that disabled because Jetpack will not handle content of Newsletters posts?

Reply:
This was an assumption on my part. When it's enabled, it only shows other newsletters, so I felt it was less useful than recirc on regular posts. Do you think publishers would want Related Posts to appear on newsletters if they only show other newsletters?

Reply:
I don't have a strong opinion here, I think it wouldn't hurt. Plus, a use case for this public newsletters posts feature is to allow a potential subscriber to assess what kind of content appears in the newsletters – a Related Posts section would be a nice way of navigating to other newsletter posts to get a better overview.

Reply:
I think you've convinced me that we shouldn't disable by default. I added an option in cbaa4d3 that lets editors decide one way or another—what do you think?